remote-api/src/main/java/org/alfresco/rest/api/probes/ProbeEntityResource.java

```diff
@@ -41,123 +43,104 @@
-    public String doCheckOrNothing(boolean isLiveProbe)
+    public String doReadyCheck()
     {
         boolean doCheck = false;
         long now = 0;
-        boolean result;
+        boolean result = false;
         String message = "No test";
         boolean logInfo = false;
-        synchronized(checking)
+        synchronized (lock)
         {
-            // Initially ready needs to be false so we don't get requests and live true so the pod is not killed.
-            if (checkResult == null)
-            {
-                result = isLiveProbe;
-            }
-            else
+
+            now = System.currentTimeMillis();
+            if (checkResult == null || isAfterCheckPeriod(now))
             {
-                result = checkResult;
+                doCheck = true;
             }
 
-            if (checking) // Is another thread is checking?
+            if (doCheck)
             {
-                if (!readySent && result && !isLiveProbe)
+                try
                 {
-                    readySent = true;
-                    logInfo = true;
+                    message = "Tested";
+                    performReadinessCheck();
+                    result = true;
                 }
-            }
-            else // This thread will do a check
-            {
-                now = System.currentTimeMillis();
-                if (checkResult == null || nextCheckTime <= now)
+                catch (Exception e)
                 {
-                    doCheck = true;
-                    checking = true;
+                    result = false;
                 }
-            }
-        }
-
-        if (doCheck)
-        {
-            try
-            {
-                message = "Tested";
-                doCheck(isLiveProbe);
-                result = true;
-            }
-            catch (Exception e)
-            {
-                result = false;
-            }
-            finally
-            {
-                synchronized (checking)
+                finally
                 {
-                    checking = false;
+
                     checkResult = result;
-                    setNextCheckTime(now);
-                    if (result && !readySent && !isLiveProbe) // Are we initially ready
-                    {
-                        readySent = true;
-                        logInfo = true;
-                    }
-                    else if (!result && (isLiveProbe || readySent)) // Are we sick
-                    {
-                        logInfo = true;
-                    }
+                    setLastCheckTime(now);
+                    logInfo = true;
+
                 }
             }
+            else
+            {
+                // if no check is performed, use previous check result
+                result = checkResult;
+
+            }
         }
 
-        message = getMessage(isLiveProbe, result, message);
+        message = getMessage(false, result, message);
```

Comment:
If we get rid of the `getMessage` method then we can also use slf4j tokens to avoid string concatenation when debug logging is turned off:
```
LOGGER.debug("readyProbe: Success - {}", message);
```